common/src/main/cpp/classes/openxr_fb_scene_manager.cpp

```diff
@@ -0,0 +1,391 @@
+#include <godot_cpp/classes/xr_server.hpp>
+
+#include "extensions/openxr_fb_scene_capture_extension_wrapper.h"
+#include "extensions/openxr_fb_scene_extension_wrapper.h"
+#include "classes/openxr_fb_spatial_entity_query.h"
+
+using namespace godot;
+
+void OpenXRFbSceneManager::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_default_scene", "packed_scene"), &OpenXRFbSceneManager::set_default_scene);
+	ClassDB::bind_method(D_METHOD("get_default_scene"), &OpenXRFbSceneManager::get_default_scene);
+
+	ClassDB::bind_method(D_METHOD("set_scene_setup_method", "method_name"), &OpenXRFbSceneManager::set_scene_setup_method);
+	ClassDB::bind_method(D_METHOD("get_scene_setup_method"), &OpenXRFbSceneManager::get_scene_setup_method);
+
+	ClassDB::bind_method(D_METHOD("set_auto_create", "enable"), &OpenXRFbSceneManager::set_auto_create);
+	ClassDB::bind_method(D_METHOD("get_auto_create"), &OpenXRFbSceneManager::get_auto_create);
+
+	ClassDB::bind_method(D_METHOD("set_visible", "visible"), &OpenXRFbSceneManager::set_visible);
+	ClassDB::bind_method(D_METHOD("get_visible"), &OpenXRFbSceneManager::get_visible);
+	ClassDB::bind_method(D_METHOD("show"), &OpenXRFbSceneManager::show);
+	ClassDB::bind_method(D_METHOD("hide"), &OpenXRFbSceneManager::hide);
+
+	ClassDB::bind_method(D_METHOD("create_scene_anchors"), &OpenXRFbSceneManager::create_scene_anchors);
+	ClassDB::bind_method(D_METHOD("remove_scene_anchors"), &OpenXRFbSceneManager::remove_scene_anchors);
+	ClassDB::bind_method(D_METHOD("are_scene_anchors_created"), &OpenXRFbSceneManager::are_scene_anchors_created);
+
+	ClassDB::bind_method(D_METHOD("request_scene_capture", "request"), &OpenXRFbSceneManager::request_scene_capture, DEFVAL(""));
+	ClassDB::bind_method(D_METHOD("is_scene_capture_enabled"), &OpenXRFbSceneManager::is_scene_capture_enabled);
+
+	ClassDB::bind_method(D_METHOD("get_anchor_uuids"), &OpenXRFbSceneManager::get_anchor_uuids);
+	ClassDB::bind_method(D_METHOD("get_anchor_node", "uuid"), &OpenXRFbSceneManager::get_anchor_node);
+	ClassDB::bind_method(D_METHOD("get_spatial_entity", "uuid"), &OpenXRFbSceneManager::get_spatial_entity);
+
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "default_scene", PROPERTY_HINT_RESOURCE_TYPE, "PackedScene"), "set_default_scene", "get_default_scene");
+	ADD_PROPERTY(PropertyInfo(Variant::STRING_NAME, "scene_setup_method", PROPERTY_HINT_NONE, ""), "set_scene_setup_method", "get_scene_setup_method");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "auto_create", PROPERTY_HINT_NONE, ""), "set_auto_create", "get_auto_create");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "visible", PROPERTY_HINT_NONE, ""), "set_visible", "get_visible");
+
+	ADD_SIGNAL(MethodInfo("scene_anchor_created", PropertyInfo(Variant::Type::OBJECT, "scene_node"), PropertyInfo(Variant::Type::OBJECT, "spatial_entity")));
+	ADD_SIGNAL(MethodInfo("scene_data_missing"));
+	ADD_SIGNAL(MethodInfo("scene_capture_completed", PropertyInfo(Variant::Type::BOOL, "success")));
```

Comment:
This is redefining a signal that's already present in `OpenXRFbSceneCaptureExtensionWrapper`, but including an additional parameter.

For this and the other signals in this PR, can you instead make the signal names more explicit (e.g: `openx_fb_scene_capture_completed`).

Reply:
Once this node works properly, that is the way users should be using this. We should strive to make this good enough so we don't need to expose the wrapper classes to Godot.

If we do decide to keep that status quo, I agree with Fredia here, the signals on the wrapper should be more explicit.

Reply:
What is the advantage of making these super long signal names?

The class name is already `OpenXRFbSceneManager` so just having `OpenXRFbSceneManager.scene_capture_completed` seems like an OK signal name. In most native Godot nodes, the signal names aren't prefixed with some part of the class name. I see that we've been doing this on other classes in 'godot_openxr_vendors', but what's the rationale for it?

Reply:
For Godot native nodes, I agree that the signal names can (and should) be succinct. For vendor extensions on the other hand, we're making the signal names explicit for a couple of reasons:
- provide clear context on the origin and role of the signal: whether it's during development, or while reading documentation / guides / tutorial, this provides clear context to the developer that the functionality is vendor specific. This is especially relevant to apps / games that target multiple platforms as they may be using different set of signals depending on the platform they're running on.
- Unlike Godot native nodes, we make no backward compatibility promises on the vendor extensions. Instead we strive to deprecate them in favor of core or multi-vendor extensions the moment those are available. Making the signal names explicit makes it easy for developers to identify logic that should be updated because it's deprecated (especially bearing in mind that not all devs use the Godot Editor as their default dev environment). This PR has a good a example of what I describe; the `OpenXRFbSceneCaptureExtensionWrapper.scene_capture_completed` signal is technically being deprecated (as we don't intend for users to use the wrapper class) and replaced by the `OpenXRFbSceneManager.scene_capture_completed` signal, which also adds an additional parameter. As a dev making the switch to `OpenXRFbSceneManager`, it would be easy to forget updating the logic tied to the signal given the similarities between the old and the new. 
  - E.g: This is the code being removed in the demo; note that signal connection uses `scene_capture_completed` and not `OpenXRFbSceneManager.scene_capture_completed`
```
var scene_capture := Engine.get_singleton("OpenXRFbSceneCaptureExtensionWrapper")
if scene_capture:
	scene_capture.connect("scene_capture_completed", _on_scene_capture_completed)
```

Reply:
The pattern so far has been to use part of the class name. For example `openxr_fb_hand_tracking_mesh_data_fetched` for the `OpenXRFbHandTrackingMeshExtensionWrapper` class.

Reply:
Signal names update in my latest push!